src/settings/storeconf.cpp

```diff
@@ -9,126 +9,165 @@
 #include <QFileDialog>
 #include <QApplication>
 
-StoreConf::StoreConf(QWidget * parent)
-    : QWidget (parent)
+StoreConf::StoreConf(QString userName)
+    : mConFile {userName += ".json"}
 {
-    conFile = getPathToConFile();
-    if (!conFile.size()) // conf.json file not exist. keep all fields with default values,
-    {                   // create conf.json with these values
+    mFields["applicationName"] = "Pair Storm";       //  default values used when json file
+    mFields["applicationVersion"] = "0.1";           //      not exist or corrupted
+    mFields["organizationName"] = "Lv-425.C++";
+    mFields["style"] = "WHITE";
+    mFields["editorFontName"] = "Consolas";
+    mFields["editorFontSize"] = "12";
+    mFields["defaultUdpPortNumber"] = "36108";
+    mFields["defaultTcpPortNumber"] = "32807";
+    mFields["userName"] = "unnamed user";
```

Comment:
Why do you pass "unnamed user" to **mFields[**"username"**]** when you have **userName** parameter.
Think about typing in parameter list **QString userName = "unnamed user"** and in the code **mFields[**"userName**] = userName;

Reply:
userName is a key
key should be common for all cases

Reply:
changed to userName